src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -128,18 +129,26 @@ def find_subscriptions_on_login(self,  # pylint: disable=too-many-arguments
                 subscriptions = subscription_finder.find_from_user_account(
                     username, password, tenant, self._ad_resource_uri)
 
-        if not subscriptions:
-            raise CLIError('No subscriptions found for this account.')
+        if not allow_no_subscriptions and not subscriptions:
+            raise CLIError("No subscriptions were found for '{}'. If this is expected, use "
+                           "'--allow-no-subscriptions' to have a tenant level account".format(
+                               username))
```

Comment:
Why do we need an explicit flag for this?  If there are subscriptions but --allow-no-subscriptions is specified, what happens?  If the behavior isn't changed, then why wouldn't we just issue a warning and proceed with the `--allow-no-subcriptions` logic?